### PR TITLE
Add version prints to ResourceManager for logging/replay

### PR DESCRIPTION
**Pull request by davidbeckingsale**
Example output for logging:

```
$ UMPIRE_LOG_LEVEL=Debug ./bin/examples/strategy_example
[DEBUG][/Users/beckingsale1/Projects/umpire/code/umpire/src/umpire/ResourceManager.cpp:77]: ResourceManager () entering
[DEBUG][/Users/beckingsale1/Projects/umpire/code/umpire/src/umpire/ResourceManager.cpp:113]: initialize () entering
[DEBUG][/Users/beckingsale1/Projects/umpire/code/umpire/src/umpire/ResourceManager.cpp:117]: initialize Umpire v0.3.2
[DEBUG][/Users/beckingsale1/Projects/umpire/code/umpire/src/umpire/ResourceManager.cpp:187]: initialize () leaving
[DEBUG][/Users/beckingsale1/Projects/umpire/code/umpire/src/umpire/ResourceManager.cpp:107]: ResourceManager () leaving
```

and for replay:

```
$ UMPIRE_REPLAY=On  ./bin/examples/strategy_example
REPLAY,Umpire v3.3.2,
REPLAY,makeMemoryResource,HOST,0x7fb1c7c03f60
```

Note that the logging print has the version number printing in `ResourceManager::initialize` - this was the easiest place I could think to print it once, but I'm open to suggestions.

### CODE REVIEW

**Comment by mcfadden8** [comment]
I think that the replay program will need to be modified to look for, and ignore, the REPLAY,Umpire... line.  Otherwise, replay will treat it as an error.

Now that we have the version information, we are able to do backwards compatibility changes.

File: tools/replay.cpp [hunk 1]
```diff
@@ -134,6 +134,8 @@ class Replay {
           replay_release();
           replay_out() << "\n";
         }
+        else if ( m_row[1] == "Umpire v0.3.2") {
```
Comment (mcfadden8) [inline tools/replay.cpp]: Hmmm.  Might want to change this to something more general.  Otherwise, we will need to update the tool for each new version.

Comment (davidbeckingsale) [reply tools/replay.cpp]: Good point 😄 I'll do that now.

Comment (davidbeckingsale) [reply tools/replay.cpp]: @mcfadden8 - this should be resolved.

**Pull request merged**